### DISCUSSION TITLE
Why Eloquent Builder methods returns static?

### DISCUSSION THREAD

**Discussion by localusercamp** [laravel/framework#46291] [Q&A]
I found that Illuminate\Database\Eloquent\Builder methods `@return` phpDoc annotations contains both `Model` and `static` type or (`Collection` and `static[]`). Why it so? When builder can return array of queries from `get()` call?

![image](https://user-images.githubusercontent.com/45385480/221791638-a163ca33-195d-410e-a630-d691545341c4.png)

**Comment by veelasky** [comment]
`static[]` return type, infer to an array containing one or many 'static' model instance. as to when builder return an array, the `Collection` instance is basically an array, just supercharged, because you can treat `Collection` as a regular array thanks to its implementation of `ArrayAccess` interface

**Reply by localusercamp** [reply]
As far as I know `static` is a reference to the class in which it is used or any of its successors.

> static[] return type, infer to an array containing one or many 'static' model instance

But `static` keyword in `Illuminate\Database\Eloquent\Builder` class means `Illuminate\Database\Eloquent\Builder` class or any of its successors. So it is **not** the array of models it is array of **Queries** (Builder instances).

Am i wrong?

**Reply by localusercamp** [reply]
If i'm right then how can `get()` return an array of Builder instances?

**Reply by mikeydevelops** [reply]
@localusercamp  you are correct, static refers to the underlying class and its children. `get` in this case will never return a builder instance. So I thing this is leftover from some older version of laravel.

**Reply by localusercamp** [reply]
@mikeydevelops How we can prove, or disaprove your thoughts? =D

**Reply by localusercamp** [reply]
I found [this commit](https://github.com/laravel/framework/commit/60995f6d3a97a55cc387fa0cb758311cd0d53d3d). 
I don't know how it looked at the time of the commit, but now it looks strange in my opinion. For me it ruins IDE support (VScode). Should this be removed?

**Reply by veelasky** [reply]
> As far as I know `static` is a reference to the class in which it is used or any of its successors.
> 
> > static[] return type, infer to an array containing one or many 'static' model instance
> 
> But `static` keyword in `Illuminate\Database\Eloquent\Builder` class means `Illuminate\Database\Eloquent\Builder` class or any of its successors. So it is **not** the array of models it is array of **Queries** (Builder instances).
> 
> Am i wrong?

yes, you're right.

**Comment by localusercamp** [comment]
I opened an issue https://github.com/laravel/framework/issues/46292
